### PR TITLE
Product store

### PR DESCRIPTION
**Pull request by lvbn**
- real user and product ids are sent to the BE
- Shopping cart items are fetched from DB

### CODE REVIEW

**Review by jamalbran — APPROVED** [review]

**Review by sherrilla71940 — APPROVED** [review]

File: client/src/services/shopping-cart-service.tsx [hunk 1]
```diff
@@ -47,10 +47,29 @@ export async function getShoppingCartProducts(body: any) {
...
+    }
+
+  } catch (error) {
+    console.error(error)
+  }
+}
+
+export async function getShoppingCartProduct(body: any) {
+  try {
+    const response = await fetch('http://localhost:3000/shoppingcartproduct', {
```
Comment (karoteblurs) [inline client/src/services/shopping-cart-service.tsx:64]: the string local host should be saved in an external variable, so at the end we packed all those in an env file

File: client/src/components/item-details/ItemDetails.tsx [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
+import { ProductSize } from '../../models/models'
 import { useState } from 'react'
 import { addToShoppingCart } from '../../services/shopping-cart-service'
+import { useProductsSlice } from '../../zustand/ProductSlice'
+import { userStore } from '../../zustand/UserStore'
+import { ShoppingCartProductType } from '../../../../global-types/shopping-cart-product'
 
 
 export default function ItemDetails() {
+  const id = userStore((state) => state.id)
```
Comment (karoteblurs) [inline client/src/components/item-details/ItemDetails.tsx:15]: consider using object destructuring to make the code less verbose:
const { id } = userStore() gives the same result.

File: client/src/components/item-details/ItemDetails.tsx [hunk 3]
```diff
@@ -28,7 +35,7 @@ export default function ItemDetails() {
     // console.log('PRODUCT:', {...product, size: size})
     // return {...product, size: size}
     if (size && product) {
-      setItem({...product, size: size})
+      // setItem({...product, size: size})
 
       // change this later to useRef
       const sizes = document.querySelectorAll('.size')
```
Comment (karoteblurs) [inline client/src/components/item-details/ItemDetails.tsx:41]: we can't have a mix of jquery and react in the app.

**Review by karoteblurs — DISMISSED** [review]

File: client/src/services/shopping-cart-service.tsx (hunk 1 above)
Comment (lvbn) [reply client/src/services/shopping-cart-service.tsx:64]: right, we can do this later

File: client/src/components/item-details/ItemDetails.tsx (hunk 2 above)
Comment (lvbn) [reply client/src/components/item-details/ItemDetails.tsx:15]: Thanks for the tip, will consider

File: client/src/components/item-details/ItemDetails.tsx (hunk 3 above)
Comment (lvbn) [reply client/src/components/item-details/ItemDetails.tsx:41]: We don't, unless somebody else is doing it. Do you see any jquery packages installed?

**Review by karoteblurs — APPROVED** [review]

**Pull request merged**